### PR TITLE
Support adding aliases to Query Blocks closes #30

### LINKED ISSUE #30: Support adding aliases to Query Blocks
Then the QUERYBUILDER smartblock command should be able to reference these aliases from the block or from the page name

### PR DESCRIPTION
**Pull request by mdroidian**

### CODE REVIEW

**Comment by mdroidian** [comment]
doesn't deal with use case of multiple same named queries (just takes the first one)

- some block
  - `{{query block:myQuery}}`
- some other block 
  - `{{query block:myQuery}}`

Page: `queries/myQuery`

File: src/index.ts [hunk 1]
```diff
@@ -422,8 +422,20 @@ export default runExtension({
...
+                 [and [?b :block/string ?s] [[clojure.string/includes? ?s "{{query block:${queryRef}}}"]] ]
+                 ${getQueryPages(extensionAPI).map(
+                   (p) =>
+                     `[and [?b :node/title ?t] [[re-pattern "^${p.replace(
+                       /\*/,
+                       queryRef
+                     )}$"] ?regex] [[re-find ?regex ?t]]]`
+                 )}
+            ]]`
+          )[0]?.toString();
```
Comment (dvargas92495) [inline src/index.ts:437]: I was hoping for the alias to be editable in the UI but that may not be necessary. At the very least, it should be viewable in the UI somehow. If it's editable, we would have to build out:
```
- {{query block}}
    - alias
        - example
```

which might be more trouble than it's worth. Your solution handles the issue rather quickly. Thoughts?

In terms of retrieval, this is 🎯

File: src/index.ts [hunk 2]
```diff
@@ -422,8 +422,20 @@ export default runExtension({
...
+          const queryRef = variables[arg] || arg;
+          const aliasOrPage = window.roamAlphaAPI.data.fast.q(
+            `[:find ?uid :where [?b :block/uid ?uid] [or-join [?b] 
+                 [and [?b :block/string ?s] [[clojure.string/includes? ?s "{{query block:${queryRef}}}"]] ]
+                 ${getQueryPages(extensionAPI).map(
+                   (p) =>
+                     `[and [?b :node/title ?t] [[re-pattern "^${p.replace(
+                       /\*/,
+                       queryRef
+                     )}$"] ?regex] [[re-find ?regex ?t]]]`
```
Comment (dvargas92495) [inline src/index.ts:434]: why not a straight comparison `[and [?b :node/title "^${p.replace(/\*/, queryRef)}$"]]`?

I also think we'll want to add a branch for comparing the title directly: `[and [?b :node/title "${queryRef}"]]`

File: src/index.ts (hunk 1 above)
Comment (dvargas92495) [reply src/index.ts:437]: Reflecting on it more, I think I prefer the approach you took over the one I had in mind. Only addition I'd make is to make that alias viewable in the query block somehow

File: src/index.ts (hunk 2 above)
Comment (mdroidian) [reply src/index.ts:434]: What is the use case for comparing the title directly?

File: src/index.ts (hunk 1 above)
Comment (mdroidian) [reply src/index.ts:435]: I could see this being nice to unify where to find alias across `Query Blocks`, `Query Drawer`, and `Query Pages`!

**Comment by mdroidian** [comment]
Not super happy with the default `Query Alias`  `<span>`
It doesn't seem intuitive as to what it's function is.

I'll come back to this.

File: src/components/QueryEditor.tsx [hunk 3]
```diff
@@ -494,6 +500,38 @@ const QueryEditor: typeof window.roamjs.extension.queryBuilder.QueryEditor = ({
...
+                setIsEditingLabel(false);
+              }
+            }}
+            autoFocus
+            rightElement={
+              <Button
+                minimal
+                icon={"confirm"}
+                onClick={() => {
+                  updateBlock({ uid: parentUid, text: `{{query block:${label}}}` });
```
Comment (dvargas92495) [inline src/components/QueryEditor.tsx:521]: what does this UX look like? does the component flicker or is it smooth?

File: src/index.ts (hunk 1 above)
Comment (dvargas92495) [reply src/index.ts:435]: Agreed! Very nicely done.

File: src/components/QueryEditor.tsx [hunk 4]
```diff
@@ -494,6 +500,39 @@ const QueryEditor: typeof window.roamjs.extension.queryBuilder.QueryEditor = ({
...
+            }
+          />
+        ) : (
+          <span 
+            style={{display: "inline-block"}}
+            tabIndex={-1} 
+            onClick={() => setIsEditingLabel(true)}
+          >
+            {!!label ? label : "Query Alias"}
+          </span>
```
Comment (dvargas92495) [inline src/components/QueryEditor.tsx:534]: > Not super happy with the default Query Alias <span>

When there's no alias, what about a span the is italicized, low opacity, lower font, that says `Click to edit alias` ?

File: src/components/QueryEditor.tsx (hunk 3 above)
Comment (mdroidian) [reply src/components/QueryEditor.tsx:521]: Seems smooth.

https://user-images.githubusercontent.com/3792666/219730603-51b4925c-158c-4eab-96a1-4d339b28d82f.mp4

File: src/components/QueryEditor.tsx (hunk 4 above)
Comment (mdroidian) [reply src/components/QueryEditor.tsx:534]: Yeah, not bad!

image

I was thinking that if majority of users / queries won't use this feature then to move it to the menu.
But then the user would have to click once (or twice) to see the alias.

But I think the muted text is probably acceptable.

File: src/index.ts (hunk 1 above)
Comment (mdroidian) [reply src/index.ts:435]: Let me know if you want that done in this PR or if we should ship what we have and look at unifying it later.

I've converted the PR from a draft either way.

Comment (dvargas92495) [reply src/index.ts:435]: no need for it to be done now this looks good to ship

**Review by dvargas92495 — APPROVED** [review]
:fire:

**Pull request merged**